tests/components/devolo_home_network/test_switch.py

```diff
@@ -0,0 +1,286 @@
+        new=AsyncMock(),
+    ) as turn_on:
+        await hass.services.async_call(
+            PLATFORM, SERVICE_TURN_ON, {"entity_id": state_key}, blocking=True
+        )
+
+        state = hass.states.get(state_key)
+        assert state is not None
+        assert state.state == STATE_ON
+        turn_on.assert_called_once_with(True)
+
+    await hass.config_entries.async_unload(entry.entry_id)
+
+
+@pytest.mark.parametrize(
+    "name, get_method, update_interval",
+    [
+        ["enable_guest_wifi", "async_get_wifi_guest_access", SHORT_UPDATE_INTERVAL],
+        ["enable_leds", "async_get_led_setting", SHORT_UPDATE_INTERVAL],
+    ],
+)
+async def test_device_failure(
+    hass: HomeAssistant,
+    mock_device: MockDevice,
+    name: str,
+    get_method: str,
+    update_interval: timedelta,
+):
+    """Test device failure."""
+    entry = configure_integration(hass)
+    device_name = entry.title.replace(" ", "_").lower()
+    state_key = f"{PLATFORM}.{device_name}_{name}"
+
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(state_key)
+    assert state is not None
+
+    api = getattr(mock_device.device, get_method)
+    api.side_effect = DeviceUnavailable
+    async_fire_time_changed(hass, dt.utcnow() + update_interval)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(state_key)
+    assert state is not None
+    assert state.state == STATE_UNAVAILABLE
+
+
+@pytest.mark.parametrize(
+    "name, set_method",
+    [
+        ["enable_guest_wifi", "async_set_wifi_guest_access"],
+        ["enable_leds", "async_set_led_setting"],
+    ],
+)
+async def test_auth_failed(
+    hass: HomeAssistant, mock_device: MockDevice, name: str, set_method: str
+):
+    """Test setting unautherized triggers the reauth flow."""
+    entry = configure_integration(hass)
+    device_name = entry.title.replace(" ", "_").lower()
+    state_key = f"{PLATFORM}.{device_name}_{name}"
+
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(state_key)
+    assert state is not None
+
+    setattr(mock_device.device, set_method, AsyncMock())
+    api = getattr(mock_device.device, set_method)
+    api.side_effect = DevicePasswordProtected
+
+    await hass.services.async_call(
+        PLATFORM, SERVICE_TURN_ON, {"entity_id": state_key}, blocking=True
+    )
+    flows = hass.config_entries.flow.async_progress()
+    assert len(flows) == 1
+
+    flow = flows[0]
+    assert flow["step_id"] == "reauth_confirm"
+    assert flow["handler"] == DOMAIN
+    assert "context" in flow
+    assert flow["context"]["source"] == SOURCE_REAUTH
+    assert flow["context"]["entry_id"] == entry.entry_id
+
+    await hass.services.async_call(
+        PLATFORM, SERVICE_TURN_OFF, {"entity_id": state_key}, blocking=True
+    )
+    flows = hass.config_entries.flow.async_progress()
+    assert len(flows) == 2
```

Comment:
This looks weird. There should only be one active reauth flow for the same config entry.